src/Datasource/RulesChecker.php

```diff
@@ -327,6 +327,13 @@ protected function _addError($rule, $name, $options)
                 $message = [$message];
             }
             $entity->errors($options['errorField'], $message);
+
+            $invalid = null;
+            if (isset($entity->{$options['errorField']})) {
+                $invalid = $entity->{$options['errorField']};
+            }
+            $entity->invalid($options['errorField'], $invalid);
```

Comment:
This line should be moved inside the if, where the check for the array property was done